### PR TITLE
Start using GitHub Actions to run builds

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

Start using GitHub Action to see if it's better/faster than CircleCI. This is a minimal change, we may wish to do more, e.g. publish the build image as a versioned action.

We might want to use GitHub Action for contributor workflow automation as well. Let's see how well it works :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,6 +46,11 @@ INTEGRATION_TEST_ARGS += -eksctl.version=$(INTEGRATION_TEST_VERSION)
 $(info will launch integration tests for Kubernetes version $(INTEGRATION_TEST_VERSION))
 endif
 
+.PHONY: github-build
+github-build:
+	cp -a . $(GOPATH)/src/$(import_prefix)
```
Comment (errordeveloper) [inline Makefile]: @2opremio I believe we won't have to do this, if we switch to modules...

Comment (2opremio) [reply Makefile]: Exactly, we won't :)

File: Makefile [hunk 2]
```diff
@@ -45,6 +46,11 @@ INTEGRATION_TEST_ARGS += -eksctl.version=$(INTEGRATION_TEST_VERSION)
 $(info will launch integration tests for Kubernetes version $(INTEGRATION_TEST_VERSION))
 endif
 
+.PHONY: github-build
+github-build:
```
Comment (2opremio) [inline Makefile]: Can you move this to the workflow file?

**Comment by 2opremio** [comment]
What's the plan/criteria to transition to the GitHub Actions? (i.e. do we want to run both in parallel for a while? what would make us switch?).

In terms of speed, GitHub runs 40% slower (it took ~10.5 minutes vs CircleCI's ~7.45 minutes , which may be an outlier from a single run, but still),  which isn't great. The feedback loop should be fast in PRs. In fact, CircleCI is already very slow (which I plan to address after #674 by persisting the Go build cache between runs), particularly considering we want to include a smoke test.

Any idea of why GitHub is slower and whether it's easy to improve? Does it support caching? (asking since I have never used it myself).

**Comment by stefanprodan** [comment]
@2opremio there is no caching support in GH actions. You can share files between actions in the same workflow but not between workflow runs.

File: Makefile (hunk 2 above)
Comment (errordeveloper) [reply Makefile]: Yeah, with modules it should be just one line, so we won't need to wrap anything.

**Comment by errordeveloper** [comment]
I just think it's a nice feature of GitHub to try, and we should be able to automate more workflows etc. They also have a Docker image registry coming out soon, in theory it could all be very integrated nicely without a 2rd party CI (less clicking, if nothing else).

**Comment by errordeveloper** [comment]
Closing for now

**Pull request closed without merging**